### PR TITLE
121 chats api

### LINKED ISSUE #121: chats api
### Chat module api

- Create new chat entities.
- Create messages. 
- And add simples api`s. (Get chat, Create chat)

### PR DESCRIPTION
**Pull request by maryanchuk04**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/users/maryanchuk04/projects/3)

## Code reviewers

- [ ] @Vova-Romaniuk 
- [ ] @Serhiy-dotcom 

### Second Level Review

- [ ] @Varvara-Maria 

## Summary of issue

Create ability write messages in chat

## Summary of change

Create new API endpoints. Write chat hub with signalR

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: StudCity.Frontend/src/components/AuthRoute/index.jsx [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
 import { tokenProtection } from '../../pages/routeProtection'
 import Layout from '../Layout'
+import { useDispatch, useSelector } from 'react-redux'
+import { connectToChatHub, selectHubConnection } from '../../app/features/chatsSlice'
 
 const AuthRoute = ({ component }) => {
+	const dispatch = useDispatch();
+	const hubConnection = useSelector(selectHubConnection);
+
+
```
Comment (Vova-Romaniuk) [inline StudCity.Frontend/src/components/AuthRoute/index.jsx:12]: delete this row

File: StudCity.Frontend/src/components/Sidenav/index.jsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ const Sidenav = () => {
 		<div className='w-auto max-w-xl flex h-full'>
 			<IconNavbar />
 			{isOpen && <FullNavbar active={active} activeName={activeName} />}
+
```
Comment (Vova-Romaniuk) [inline StudCity.Frontend/src/components/Sidenav/index.jsx:13]: and this

File: StudCity.Frontend/src/services/hubService.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+const tokenService = new TokenService();
+
+export const connectToHub = () => {
+	if (tokenService.getToken()) {
+		const signalR = new HubService();
+		signalR.startConnection(() => console.log("START_CONNECTION"));
+		return signalR;
+	}
+}
+
```
Comment (Vova-Romaniuk) [inline StudCity.Frontend/src/services/hubService.js:16]: here delete

**Review by Vova-Romaniuk — APPROVED** [review]

**Review by Serhiy-dotcom — APPROVED** [review]
Well done!

**Pull request merged**